app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    result_store = {}
+
+    def wrapper(*args: any) -> any:
+        if args in result_store:
+            print("Getting from cache")
+            return result_store[args]
+        else:
+            print("Calculating new result")
+            res = func(*args)
+            result_store[args] = res
```

Comment:
try to avoid creating additional variable `res` -> move return outside of a condition to avoid two returns